src/main/java/gg/uhc/uhc/inventory/IconInventory.java

```diff
@@ -68,14 +96,7 @@ protected int add(IconStack icon) {
         // add item to list in the correct location
         final int index = indexToInsert(icon);
         icons.add(index, icon);
-
-        // make sure this is enough space now
-        ensureInventorySize();
-
-        // rerender all items past this one (inclusive) as they have shifted index right 1
-        for (int i = index; i < icons.size(); i++) {
-            inventory.setItem(i, icons.get(i));
-        }
+        reopenForCurrentViewers();
```

Comment:
I don't see how a reload would cause massive amounts of inventory updates?

The modules aren't unloaded anyways (so there are no onWeightUpdate calls during UHC#onDisable) and IconInventory#onPluginShutdown automatically closes all current viewers on plugin shutdown (which is properly fixed in #100, hence #100's dependence on this).